modules/firelens/variables.tf

```diff
@@ -2,6 +2,19 @@ variable "namespace" {
   type = string
 }
 
+variable "container_registry" {
+  type        = string
+  description = "URL to the container registry for the logging image"
+
+  # See https://github.com/wellcomecollection/platform-infrastructure/tree/master/containers
+  default = "760097843905.dkr.ecr.eu-west-1.amazonaws.com/uk.ac.wellcome"
```

Comment:
Putting these in variables suggests they might ever be something else, could they be?

Reply:
Yes.

Probably worth a brief explanation on how ECR Private and ECR Public work: rather than having a single repository that can be accessed either privately or publicly, they're two entirely separate registries. This is probably done to avoid permission screw-ups: you can't accidentally expose your private container images to the entire world. (See also: S3 buckets.)

The different registries have entirely different URL schemes, e.g.:

<img width="810" alt="Screenshot 2021-06-30 at 18 55 42" src="https://user-images.githubusercontent.com/301220/124008842-c79a3b80-d9d4-11eb-8b9c-7d9332acb529.png">


<img width="890" alt="Screenshot 2021-06-30 at 18 54 50" src="https://user-images.githubusercontent.com/301220/124008733-a6394f80-d9d4-11eb-9109-08f939653ead.png">

so the private URL would be

```
760097843905.dkr.ecr.eu-west-1.amazonaws.com/uk.ac.wellcome/fluentbit
```

and the public URL would be

```
public.ecr.aws/l7a1d1z4/fluentbit
```

Being able to swap out the container registry would let a caller choose:

* To get the logging container from our private ECR registry (if it's us, running apps in our accounts)
* To get the logging container from our public ECR registry (for the prototype storage service)
* To get the logging container from an entirely different registry

and you can see an example over here: https://github.com/wellcomecollection/dams-prototype-project/pull/1/files#diff-adc12c91fefbb328c83fe772eb3b299a63b63684e32c11a56b5c2e73c92d52d2R82-R86

This is also written with an eye to more flexibility in the storage service – right now the DAMS prototype is firing logs into an AWS Elasticsearch cluster, but I can imagine people will already have logging setups they want to use. Letting people swap out the logging container without diving into the bowels of the storage service is a step towards that.

Reply:
I was aware how the public repos worked. I think I would expect this module to hold public & private defaults and then have a variable which lets you switch between the two.

Reply:
Ah, gotcha. I did think of that, but this felt slightly more flexible – I can imagine another organisation pushing containers into their own account that they own (like we do) and wanting to change this.